populator-machinery/controller_test.go

```diff
@@ -62,19 +63,30 @@ type testCase struct {
 	populateFn func(context.Context, PopulatorParams) error
 	// Provider specific data population completeness check function, return true when data transfer gets completed.
 	populateCompleteFn func(context.Context, PopulatorParams) (bool, error)
+	// The original PVC gets deleted or not
+	pvcDeleted bool
 	// PvcPrimeMutator is the mutator function for pvcPrime
 	pvcPrimeMutator func(PvcPrimeMutatorParams) (*v1.PersistentVolumeClaim, error)
 	// Expected errors
 	expectedResult error
 	// Expected keys in the notifyMap
 	expectedKeys []string
 	// Expected objects after the test runs
-	expectedObjects vpObjects
+	expectedObjects *vpObjects
 }
 
 // vpObjects includes the objects we want to compare after tests run
 type vpObjects struct {
-	pvcPrime *corev1.PersistentVolumeClaim
+	pvc *corev1.PersistentVolumeClaim
+	// When set to true, only compare PVC Finalizers
+	onlyComparePVCFinalizers bool
+	pvcPrime                 *corev1.PersistentVolumeClaim
+	// When set to true, only compare mutate feilds
+	onlyComparePVCPrimeMutateFields bool
```

Comment:
It seems like these fields are trying to exclude some comparison. I would prefer to exclude something and still use `reflect.DeepEqual`, rather than (I think this is something we'd need to do using `cmp`, rather than `reflect`). This would make the test more resilient.

Can you at least comment on the specific test cases what you want to ignore?

Reply:
sure,I'll add comments to explain why these flags are used. I have `reflect.DeepEqual` for most tests. Only for the test which the object gets updated by the volume populator I have this flag set to true because for some reason `reflect.DeepEqual` return `false` even if all the fields print out look exactly the same.

Reply:
updated